### PR TITLE
Stencil: Fix non-3D Compiles on GPU

### LINKED ISSUE #2347: Installation error: identifier "src_zeropad" is undefined
Dear developers:
        Recently, another installation error happens as seen below
        
error2
       Thanks

### PR DESCRIPTION
**Pull request by ax3l**
Fix a regression to #2314: non-3D GPU compiles did not compile (#2347)

- [x] Add a HIP 2D Test to compilation.
- [x] Fix #2347

### CODE REVIEW

File: .github/workflows/hip.yml [hunk 1]
```diff
@@ -47,3 +47,45 @@ jobs:
...
+        cmake -S . -B build_2d \
+          -DCMAKE_VERBOSE_MAKEFILE=ON \
+          -DAMReX_AMD_ARCH=gfx900     \
+          -DWarpX_DIMS=2              \
+          -DWarpX_COMPUTE=HIP         \
+          -DWarpX_EB=ON               \
+          -DWarpX_LIB=ON              \
+          -DWarpX_MPI=ON              \
+          -DWarpX_OPENPMD=ON          \
+          -DWarpX_PRECISION=SINGLE    \
```
Comment (PhilMiller) [inline .github/workflows/hip.yml]: Given the naming, was this meant to be `DOUBLE`?

Comment (ax3l) [reply .github/workflows/hip.yml]: Thx for catching this!

**Review by PhilMiller — APPROVED** [review]

**Review by RemiLehe — APPROVED** [review]

**Comment by RemiLehe** [comment]
Interesting. How come we were able to compile for CUDA in 2D, even before this PR?

**Comment by ax3l** [comment]
Its a regression since last week due to #2314 - and we indeed did not have single GPU compile test that was not 3D yet. (Mainly due to compile time.)

**Pull request merged**